python/lsst/daf/butler/registry/interfaces/_obscore.py

```diff
@@ -0,0 +1,163 @@
+    from ._dimensions import DimensionRecordStorageManager
+
+
+class ObsCoreTableManager(VersionedExtension):
+    """An interface for populating ObsCore tables(s)."""
+
+    @classmethod
+    @abstractmethod
+    def initialize(
+        cls,
+        db: Database,
+        context: StaticTablesContext,
+        *,
+        universe: DimensionUniverse,
+        config: Mapping,
+        datasets: Type[DatasetRecordStorageManager],
+        dimensions: DimensionRecordStorageManager,
+    ) -> ObsCoreTableManager:
+        """Construct an instance of the manager.
+
+        Parameters
+        ----------
+        db : `Database`
+            Interface to the underlying database engine and namespace.
+        context : `StaticTablesContext`
+            Context object obtained from `Database.declareStaticTables`; used
+            to declare any tables that should always be present in a layer
+            implemented with this manager.
+        universe : `DimensionUniverse`
+            All dimensions known to the registry.
+        config : `dict` [ `str`, `Any` ]
+            Configuration of the obscore manager.
+        datasets : `type`
+            Type of dataset manager.
+        dimensions: `DimensionRecordStorageManager`
+            Manager for Registry dimensions.
+
+        Returns
+        -------
+        manager : `ObsCoreTableManager`
+            An instance of a concrete `ObsCoreTableManager` subclass.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def config_json(self) -> str:
+        """Dump configuration in JSON format.
+
+        Returns
+        -------
+        json : `str`
+            Configuration serialized in JSON format.
+        """
+        raise NotImplementedError()
+
+    def add_datasets(self, refs: Iterable[DatasetRef]) -> None:
+        """Possibly add datasets to the obscore table, this method should be
+        called when new datasets are added to a RUN collection.
+
+        Parameters
+        ----------
+        refs : `iterable` [ `DatasetRef` ]
+            Dataset refs to add. Dataset refs have to be completely expanded.
+            If a record with the same dataset ID is already in obscore table,
+            the dataset is ignored.
+
+        Notes
+        -----
+        Dataset data types and collection names are checked against configured
+        list of collections and dataset types, non-matching datasets are
+        ignored and not added to the obscore table.
+
+        When configuration parameter ``collection_type`` is not "RUN", this
+        method should return immediately.
+
+        Note that there is no matching method to remove datasets from obscore
+        table, we assume that removal happens via foreign key constraint to
+        dataset table with "ON DELETE CASCADE" option.
+        """
+        raise NotImplementedError()
+
+    def associate(self, refs: Iterable[DatasetRef], collection: Optional[str] = None) -> None:
```

Comment:
For these methods I only need the name of a collection to match it against the configured tagged collection name. But I can make them to accept a CollectionRecord, just in case it will need to do something more complicated.

Reply:
Ah, if you don't actually need it for anything else then I don't have a strong opinion on this.